### PR TITLE
Fix SMS Install

### PR DESCRIPTION
**Pull request by wpride**
Looks like SMS install broke at some point when the requirements for a CommCareTask changed. This fixes that. Also adds the spongy castle algorithm to the proguard keep file since those were being eaten. 

Don't think we need a hotfix unless @gcapalbo disagrees

Needs https://github.com/dimagi/commcare-android/pull/1406

### CODE REVIEW

File: app/proguard.cfg [hunk 1]
```diff
@@ -46,6 +47,7 @@
 -keep,includedescriptorclasses class org.odk.collect.android.** { *; }
 -keep,includedescriptorclasses class org.javarosa.** { *; }
 -keep class in.uncod.android.bypass.** { *; }
+-keep class org.spongycastle.** { *; }
```
Comment (wpride) [inline app/proguard.cfg]: @amstone326 this is the line that solves the NoSuchAlgorithmException issue

**Comment by gcapalbo** [comment]
(offlined, answer is yes we should hotfix)

File: app/src/org/commcare/activities/CommCareSetupActivity.java [hunk 2]
```diff
@@ -607,6 +607,7 @@ protected void deliverError(CommCareSetupActivity receiver, Exception e) {
                                     }
                                 }
                             };
+                    mTask.connect(this);
```
Comment (phillipm) [inline app/src/org/commcare/activities/CommCareSetupActivity.java]: If you change `RetrieveParseVerifyMessageTask mTask` to `RetrieveParseVerifyMessageTask<CommCareSetupActivity> mTask` it gets rid of the lint warning here

File: app/src/org/commcare/tasks/RetrieveParseVerifyMessageTask.java [hunk 3]
```diff
@@ -18,9 +18,9 @@ public RetrieveParseVerifyMessageTask(RetrieveParseVerifyMessageListener listene
     }
 
     @Override
-    protected String doTaskBackground(String[] params) {
+    protected String doTaskBackground(Object... params) {
         try {
-            String url = SigningUtil.trimMessagePayload(params[0]);
+            String url = SigningUtil.trimMessagePayload(params[0].toString());
```
Comment (phillipm) [inline app/src/org/commcare/tasks/RetrieveParseVerifyMessageTask.java]: I don't understand why you have to pass this in as an `Object`

**Comment by amstone326** [comment]
@wpride So is this getting hotfixed?

**Comment by wpride** [comment]
@amstone326 yes but its not urgent and there's still one bug I'm working out

**Comment by amstone326** [comment]
Got it sounds good

**Pull request merged**